src/EnergyPlus/DXCoils.cc

```diff
@@ -2979,895 +2979,978 @@ void GetDXCoils(EnergyPlusData &state)
-
-        thisDXCoil.CrankcaseHeaterCapacity = Numbers(11);
-        if (thisDXCoil.CrankcaseHeaterCapacity < 0.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be >= 0.0  entered value=[{:.1T}].", cNumericFields(11), Numbers(11)));
-            ErrorsFound = true;
-        }
+            cFieldName = "Rated Condenser Water Flow Rate";
+            thisDXCoil.RatedHPWHCondWaterFlow = s_ip->getRealFieldValue(fields, schemaProps, "rated_condenser_water_flow_rate"); // Numbers(8);
+            // move to init
+            if (thisDXCoil.RatedHPWHCondWaterFlow != Constant::AutoCalculate) {
+                if (thisDXCoil.RatedHPWHCondWaterFlow <= 0.0) {
+                    ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
+                    ShowContinueError(state, format("...{} must be > 0.0  entered value=[{:.3T}].", cFieldName, thisDXCoil.RatedHPWHCondWaterFlow));
+                    ErrorsFound = true;
+                }
+                //   check the range of flow rate to be >= 1 gpm/ton and <= 5 gpm/ton
+                if (thisDXCoil.RatedHPWHCondWaterFlow / thisDXCoil.RatedTotCap2 < 1.79405e-8 ||
+                    thisDXCoil.RatedHPWHCondWaterFlow / thisDXCoil.RatedTotCap2 > 8.97024e-8) {
+                    ShowWarningError(state, format("{}{}=\"{}\", outside range", RoutineName, CurrentModuleObject, thisDXCoil.Name));
+                    ShowContinueError(state,
+                                      format("...{} per watt of {} is outside the recommended range of >= 1.79405E-8 m3/s/W (0.083 gpm/MBH) and <= "
+                                             "8.97024E-8 m3/s/W (0.417 gpm/MBH).",
+                                             cFieldName,
+                                             cRatedHeatingCapFieldName));
+                    ShowContinueError(
+                        state, format("...Entered Flow rate per watt = [{:.10T}].", (thisDXCoil.RatedHPWHCondWaterFlow / thisDXCoil.RatedTotCap2)));
+                }
+            }
 
-        thisDXCoil.MaxOATCrankcaseHeater = Numbers(12);
-        if (thisDXCoil.MaxOATCrankcaseHeater < 0.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be >= 0 {{C}}.  entered value=[{:.1T}].", cNumericFields(12), Numbers(12)));
-            ErrorsFound = true;
-        }
+            cFieldName = "Evaporator Fan Power Included in Rated COP";
+            std::string fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "evaporator_fan_power_included_in_rated_cop"); // Alphas(2)
+            BooleanSwitch fanPowerIncluded = static_cast<BooleanSwitch>(getYesNoValue(Util::makeUPPER(fieldValue)));
+            switch (fanPowerIncluded) {
```

Comment:
A cleaner way to do this. If you are just making a single comparison, `if` is better/faster than `switch`. Note the `ShowSevereInvalidBool` error wrapper.

```cpp
if (fanPowerIncluded != BooleanSwitch::Invalid) {
    thisDXCoil.FanPowerIncludedInCOP = static_cast<bool>(fanPowerIncluded);
} else {
     ShowSevereInvalidBool(state, eoh, cFieldName, fieldValue);
     ErrorsFound = true;
}
```

Reply:
What makes you believe that fanPowerIncluded will be anything other than Yes or No? This just seems to be overhead that's not needed. It's a direct read. No warning message, no gymnastics, just read that input to be true or false.

  A2 , \field Evaporator Fan Power Included in Rated COP
       \type choice
       \key Yes
       \key No
       \default Yes

So what would that 1 line be? I'll paste together what I see here but am probably wrong. 1 line and done.

    thisDXCoil.FanPowerIncludedInCOP = static_cast<bool>(static_cast<BooleanSwitch>(getYesNoValue(Util::makeUPPER(fieldValue))));

Reply:
This seems to be your corner. That would work. You actually don't need the `static_cast<BooleanSwitch>`.

Reply:
Done.